ord_schema/scripts/migration_v0d1d1_v0d2d2.py

```diff
@@ -0,0 +1,359 @@
+    """
+    migrate_MergeFrom_multiple(input_new, input_old, ('crude_components',))
+    input_new.addition_order = input_old.addition_order
+    migrate_CopyFrom_multiple(
+        input_new, input_old,
+        ('addition_time', 'addition_speed', 'addition_duration', 'flow_rate',
+         'addition_device', 'addition_temperature'))
+    # Changed input fields - components
+    for component_old in input_old.components:
+        component = input_new.components.add()
+        migrate_MergeFrom_multiple(component, component_old,
+                                   ('identifiers', 'preparations'))
+        if len(component_old.features) > 0:
+            raise NotImplementedError
+        if component_old.HasField('is_limiting'):
+            component.is_limiting = component_old.is_limiting
+        # Vendor information is now in a submessage
+        if (component_old.vendor_source or component_old.vendor_id or
+                component_old.vendor_lot):
+            component.source.vendor = component_old.vendor_source
+            component.source.id = component_old.vendor_id
+            component.source.lot = component_old.vendor_lot
+        # Reaction role enum is relocated and renumbered
+        role_name = component_old.ReactionRole.ReactionRoleType.DESCRIPTOR.\
+            values_by_number[component_old.reaction_role].name
+        component.reaction_role = reaction_pb2.ReactionRole.ReactionRoleType.\
+            DESCRIPTOR.values_by_name[role_name].number
+        # Amounts are nwo in a submessage
+        kind = component_old.WhichOneof('amount')
+        if kind:
+            migrate_CopyFrom(getattr(component.amount, kind),
+                             getattr(component_old, kind))
+        if component_old.HasField('volume_includes_solutes'):
+            component.amount.volume_includes_solutes = (
+                component_old.volume_includes_solutes)
+
+
+def migrate_workup(workup_new, workup_old):
+    """Copies the contents from an old workup to the new workup format.
+
+    Args:
+        workup_new: A reaction_pb2.ReactionWorkup message.
+        workup_old: A reaction_old_pb2.ReactionWorkup message.
+    """
+    type_name = workup_old.WorkupType.DESCRIPTOR.values_by_number[
+        workup_old.type].name
+    workup_new.type = workup_new.WorkupType.DESCRIPTOR.values_by_name[
+        type_name].number
+    if workup_old.HasField('input'):
+        migrate_input(workup_new.input, workup_old.input)
+    migrate_assign_multiple(workup_new, workup_old, ('details', 'keep_phase'))
+    migrate_CopyFrom_multiple(workup_new, workup_old,
+                              ('duration', 'temperature', 'stirring'))
+    if workup_old.HasField('target_ph'):
+        workup_new.target_ph = workup_old.target_ph
+    if workup_old.HasField('is_automated'):
+        workup_new.is_automated = workup_old.is_automated
+    if workup_old.HasField('mass'):
+        migrate_CopyFrom(workup_new.amount.mass, workup_old.mass)
```

Comment:
Should be `workup_new.aliquot_amount.mass`, same for volume.

Reply:
Do we need to change `aliquot_amount` -> `amount`? I've also made errors by forgetting this more specific name.

Reply:
I had fixed this in an earlier commit -- that's odd. 

When you proposed the change, I liked it because "aliquot_amount" reduces the ambiguity over what "amount" means for a workup. However, there is a chance that in the future we will add another workup type that might require the amount field, so perhaps it should be changed to "amount"?